typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -7,243 +7,237 @@ import { sleep, strip0x } from '@hyperlane-xyz/utils';
-      result = JSON.parse(responseText);
-    } catch (e) {
-      this.logger(
-        `[${chain}] Failed to parse response from ${responseText}`,
-        e,
-      );
+    // only include apikey if provided & not blockscout
+    if (family !== ExplorerFamily.Blockscout && this.apiKeys[chain]) {
+      params.set('apikey', this.apiKeys[chain]);
     }
+
+    const url = new URL(apiUrl);
+    const isGetRequest = [
+      ExplorerApiActions.CHECK_STATUS,
+      ExplorerApiActions.CHECK_PROXY_STATUS,
+      ExplorerApiActions.GETSOURCECODE,
+    ].includes(action);
+    if (isGetRequest) {
+      url.search = params.toString();
+    } else if (family === ExplorerFamily.Blockscout) {
+      // Blockscout requires module and action to be query params
+      url.searchParams.set('module', 'contract');
+      url.searchParams.set('action', action);
+    }
+
+    const response = await fetch(url.toString(), {
+      method: isGetRequest ? 'GET' : 'POST',
+      headers: isGetRequest
+        ? {}
+        : { 'Content-Type': 'application/x-www-form-urlencoded' },
+      body: isGetRequest ? null : params,
+    });
+
+    const responseText = await response.text();
+    const result = JSON.parse(responseText);
+
     if (result.message !== 'OK') {
-      const errorMessageBase = `[${chain}]`;
       let errorMessage;
 
       switch (result.result) {
         case ExplorerApiErrors.VERIFICATION_PENDING:
           await sleep(5000); // wait 5 seconds
-          return this.submitForm(chain, action, options);
+          return this.submitForm(chain, action, verificationLogger, options);
         case ExplorerApiErrors.ALREADY_VERIFIED:
         case ExplorerApiErrors.ALREADY_VERIFIED_ALT:
           return;
         case ExplorerApiErrors.PROXY_FAILED:
-          errorMessage = `${errorMessageBase} Proxy verification failed, try manually?`;
+          errorMessage = 'Proxy verification failed, try manually?';
           break;
         case ExplorerApiErrors.BYTECODE_MISMATCH:
-          errorMessage = `${errorMessageBase} Compiled bytecode does not match deployed bytecode, check constructor arguments?`;
+          errorMessage =
+            'Compiled bytecode does not match deployed bytecode, check constructor arguments?';
           break;
         default:
-          errorMessage = `${errorMessageBase} Verification failed. ${
+          errorMessage = `Verification failed. ${
             result.result ?? response.statusText
           }`;
           break;
       }
 
       if (errorMessage) {
-        this.logger(errorMessage);
-        throw new Error(errorMessage);
+        verificationLogger(errorMessage);
+        throw new Error(`[${chain}] ${errorMessage}`);
       }
     }
 
+    if (result.result === ExplorerApiErrors.UNKNOWN_UID) {
+      await sleep(1000); // wait 1 second
+      return this.submitForm(chain, action, verificationLogger, options);
+    }
+
+    if (result.result === ExplorerApiErrors.UNABLE_TO_VERIFY) {
```

Comment:
else if?

Reply:
we're returning in the previous if anyway

Reply:
I find it easier to read visually to see there isnt fall through here but fair